hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -414,71 +540,130 @@ Since the life cycle partition is the only partition that needs live updates in-
+
+Upon initialization, the controller reads out the whole partition and descrambles it on the fly if needed.
+
+Then, right after the initial readout, the partition controller jumps into the first integrity check, which behaves somewhat differently, depending on whether the partition is digest protected and scrambled.
+If the partition is not digest protected, or if the digest has not yet been computed, the check completes right away, and the buffered values are released for hardware broadcast.
+Otherwise, the partition contents in the buffer registers are re-scrambled if needed, and a digest is computed on the fly.
+If the computed digest matches with the one that has been read out before, the buffered registers are released for hardware broadcast.
+Otherwise, the buffered values are gated to their default, and an alert is triggered through the error handling logic.
+
+After initialization, the integrity check (as described above) and the consistency check can be triggered by the LFSR timer mechanism on a periodic basis.
+
+The consistency check behaves differently, depending on whether the partition is digest protected or not.
+If it is, the consistency check will read out the digest stored in OTP and compare it with the value stored in the buffer register.
+Otherwise, if no digest is available, the controller will read out the whole partition from OTP, and compare it to the contents stored in the buffer registers.
+In case of a mismatch, the buffered values are gated to their default, and an alert is triggered through the error handling logic.
+
+Note that in case of unrecoverable OTP errors or parity failures in the buffer registers, the partition controller FSM is moved into a terminal error state, which locks down all access through DAI and clamps the values that are broadcast in hardware to their defaults.
+
+### Direct Access Interface Control
+
+![Direct Access Interface FSM](otp_ctrl_dai_fsm.svg)
+
+Upon reset release, the DAI controller first sends an initialization command to the OTP macro.
+Once the OTP macro becomes operational, an initialization request is sent to all partition controllers, which will read out and initialize the corresponding buffer registers.
+The DAI then becomes operational once all partitions have initialized, and supports read, write and digest calculation commands (see [here]({{< relref "#direct-access-interface" >}}) for more information about how to interact with the DAI through the CSRs).
+
+Read and write commands transfer either 32bit or 64bit of data from the OTP to the corresponding CSR and vice versa. The access size is determined automatically, depending on whether the partition is scrambled or not. Also, (de)scrambling is performed transparently, depending on whether the partitions is scrambled or not.
+
+Digest calculation commands read out the complete contents of a particular partition, compute a digest and burn in that digest value to the predefined location at the end of the partition.
```

Comment:
Maybe "burn in" -> "write" ? I think this is the only place you use "burn in" as terminology.

Reply:
good point, I'll change it to write.